Sources/Architecture/X86/HardwareBreakpointManager.cpp

```diff
@@ -26,6 +26,16 @@ HardwareBreakpointManager::HardwareBreakpointManager(Target::Process *process)
 
 HardwareBreakpointManager::~HardwareBreakpointManager() {}
 
+ErrorCode HardwareBreakpointManager::add(Address const &address, Type type,
+                                         size_t size, Mode mode) {
+  int filled_sites = _sites.size();
+  DS2ASSERT(filled_sites <= maxWatchpoints());
+  if (filled_sites == maxWatchpoints())
```

Comment:
This is just a nit, but in general you might wanna do `if (... >= maxWatchpoints())` instead of `==` so we have a better check when we have no assertions.